src/Block/MediaBlockService.php

```diff
@@ -204,11 +208,20 @@ private function getMediaBuilder(): FormBuilderInterface
             'edit' => 'list',
         ]);
 
-        return $this->mediaAdmin->getFormBuilder()->create('mediaId', ModelListType::class, [
+        $fieldDescription->setAssociationAdmin($this->mediaAdmin);
+
+        $formBuilder = $form->getFormBuilder()->create('mediaId', ModelListType::class, [
             'sonata_field_description' => $fieldDescription,
             'class' => $this->mediaAdmin->getClass(),
             'model_manager' => $this->mediaAdmin->getModelManager(),
             'label' => 'form.label_media',
         ]);
+
+        $formBuilder->addModelTransformer(new CallbackTransformer(
+            static fn (mixed $value): ?PageBlockInterface => $value,
```

Comment:
Seems like a use PageBlockInterface need to be added

Reply:
Hi @VincentLanglet  ! Im on it today :).
Thanks for review :pray: 

PageBlockInterface is up on PageBundle, did i need require it ? or i need rewrite the logic ?

Reply:
Oh...
This would be weird requiring the PageBundle here.

Are you sure the callback works with PageBlockInterface then ?
Since it's a `mediaId` I would say it's
```
$formBuilder->addModelTransformer(new CallbackTransformer(
    static fn (mixed $value): ?MediaInterface => $value,
    static fn (?PageBlockInterface $value) => $value instanceof MediaInterface ? $value->getId() : $value
));
```

Reply:
I have to confess that I'm not sure I ever tested this on my customer project.
In any case, your solution speaks to me. I'll take a look and commit if need be!

Reply:
You're right! 
I must have missed a copy-paste while testing to move up the version.